src/agent/Core/ApplicationPool/Group/ProcessListManagement.cpp

```diff
@@ -113,11 +125,16 @@ Group::findEnabledProcessWithLowestBusyness() const {
 
 	int leastBusyProcessIndex = -1;
 	int lowestBusyness = 0;
-	unsigned int i, size = enabledProcessBusynessLevels.size();
+	unsigned int i, size = enabledProcessBusynessLevels.size(), highest_gen = 0;
 	const int *enabledProcessBusynessLevels = &this->enabledProcessBusynessLevels[0];
 
 	for (i = 0; i < size; i++) {
-		if (leastBusyProcessIndex == -1 || enabledProcessBusynessLevels[i] < lowestBusyness) {
+		if (leastBusyProcessIndex == -1 ||
+			enabledProcessBusynessLevels[i] < lowestBusyness ||
+			(enabledProcessBusynessLevels[i] == lowestBusyness && enabledProcesses[i]->generation > highest_gen)) {
```

Comment:
I think there should not be a check for `enabledProcessBusynessLevels[i] == lowestBusyness` here. Check earlier comment.